### PR TITLE
9 clean up reset function

### PR DESCRIPTION
**Pull request by ken7184**
I modified the reset function.
1. Reset action space and action spaces
2. Reset Initialize isFirst variable that other students add
3. add self.rewards part to store the immediate reward for each agent after every action in the environment.
4. Reset pygame visualization part

### CODE REVIEW

**Comment by ken7184** [comment]
@Polfish Thanks.

**Comment by Polfish** [comment]
It looks good to me.

File: ourhexenv.py [hunk 1]
```diff
@@ -57,14 +57,25 @@ def __init__(self, board_size=5, sparse_flag=True):
 
     def reset(self):
         self.board = np.zeros((self.board_size, self.board_size), dtype=int)
+        
+        self.isFirst = True
```
Comment (ChristoferBerruz) [inline ourhexenv.py:61]: Python uses snakecase instead of camelCase. Consider changing this to is_first instead

File: ourhexenv.py [hunk 2]
```diff
@@ -57,14 +57,25 @@ def __init__(self, board_size=5, sparse_flag=True):
...
         self.dones = {agent: False for agent in self.agents}
         self.infos = {agent: [] for agent in self.agents}
-
         self._cumulative_rewards = {agent: 0 for agent in self.agents}
         self.terminations = {agent: False for agent in self.agents}
         self.truncations = {agent: False for agent in self.agents}
+        self.rewards = {agent: 0 for agent in self.agents}
+
+        self.action_space = spaces.Discrete(self.board_size * self.board_size + 1)
+        self.action_spaces = {agent: self.action_space for agent in self.agents}
```
Comment (ChristoferBerruz) [inline ourhexenv.py:73]: There is no need to reset the action spaces as they remain constant and unchanged during the lifetime of the environment

**Review by ChristoferBerruz — CHANGES_REQUESTED** [review]

File: ourhexenv.py (hunk 1 above)
Comment (ken7184) [reply ourhexenv.py]: Got it thanks

File: ourhexenv.py (hunk 2 above)
Comment (ken7184) [reply ourhexenv.py]: Got it thanks. I will modify that.

**Pull request merged**